src/app/widgets/inlineinput/inlineinput.component.ts

```diff
@@ -83,12 +83,16 @@ export class InlineInputComponent implements OnInit {
     }
   }
 
+  isFloat(x) {
+    return typeof x === 'number' && Number.isFinite(x) && x >= -2147483648 && x <= 2147483648;
+  }
+
   validateValue(value) {
     if (this.type === 'integer') {
       return /^\d+$/.test(value);
     }
     if (this.type === 'float') {
-      return /^-?\d*(\.\d+)?$/.test(value);
+      return this.isFloat(value);
```

Comment:
@abhinandan13jan This method is still failing because `value` is of type `string`. `value` needs to be parsed first for this to work.

Reply:
Updated

Reply:
@sahil143 I was trying not to alter any preexisting functionalities. Now, I've added a .focus() method for the validation failure case. Can you please check again if it looks ok now?

Reply:
@joshuawilson I've updated using sematic release format